server/testutil/testolapdb/testolapdb.go

```diff
@@ -0,0 +1,62 @@
+package testolapdb
+
+import (
+	"context"
+	"sync"
+	"testing"
+
+	"github.com/buildbuddy-io/buildbuddy/server/tables"
+	"github.com/stretchr/testify/require"
+	"gorm.io/gorm"
+
+	iepb "github.com/buildbuddy-io/buildbuddy/proto/internal_execution"
+)
+
+type TestOLAPDBHandle struct {
+	executionIDsByInvID sync.Map // map of invocationID => a slice of execution IDs
+	invIDs              sync.Map // map of invocationID => struct{}
+}
+
+func NewTestOLAPDBHandle() *TestOLAPDBHandle {
```

Comment:
nit: you could maybe call this `New()` so that the callsite reads like `testolapdb.New()`

`TestOLAPDBHandle` can also be named `Handle` so it reads like `testolapdb.Handle`